Scala/Scala.sublime-syntax

```diff
@@ -440,16 +578,123 @@ contexts:
       push:
         - match: '`'
           pop: true
-    - match: '\b[a-z][a-zA-Z0-9_]*\.'
-    - match: '\.[a-z][a-zA-Z0-9_]*\b'
+    - match: '{{varid}}\.'
+    - match: '\.{{varid}}'
     - match: '\b{{varid}}'
       scope: entity.name.parameter.scala
+    - match: \[
+      push:
+        - match: \]
+          pop: true
+        - include: main
+    - match: '@|_'
+      scope: keyword.other.scala
+  val-pattern-match:
+    - match: '{{upperid}}(?=[\s=])'
+      scope: entity.name.parameter
+    - match: '{{typeprefix}}'
+      push:
+        - match: '(?=[\B\s]=[\B\s])'
+          pop: true
+        - include: delimited-type-expression
+        - match: '(?=[\{\n])'
+          pop: true
+    - include: val-pattern-match-main
+    - match: \(
+      push:
+        - match: \)
+          pop: true
+        - include: val-pattern-match-inner
+  val-pattern-match-inner:
+    - include: val-pattern-match-main
     - match: '{{upperid}}'
       scope: support.class.scala
+
+  pattern-match:
+    - include: keywords
+    - include: constants
+    - include: char-literal
+    - include: scala-symbol
+    - include: strings
+    - include: xml-literal
+    - match: '`'
```

Comment:
Ok, so the `` ` `` should probably be scoped as `punctuation.definition.identifier.scala` then. Or possibly some other `punctuation.definition.` scope.